### PR TITLE
scripts/build_utils.sh: add package repo for apache arrow (enables parquet objects processing)

### PR DESCRIPTION
**Pull request by galsalomon66**
…y) to reduce risk of aborting build process due to changes;

Signed-off-by: gal salomon <gal.salomon@gmail.com>

### CODE REVIEW

**Comment by galsalomon66** [comment]
https://github.com/ceph/ceph/pull/40802 is blocked on this PR.

reviewing the logs of previous PR(had failed)

1. both platforms need to import keys
2. curl installation is missing
3. add-apt-repository is also installed, in the case its missing it will break the build. 
4. changes had tested with container images (from scratch)

File: scripts/build_utils.sh [hunk 1]
```diff
@@ -489,6 +489,13 @@ echo "deb [arch=amd64] http://us.archive.ubuntu.com/ubuntu/ $DIST-backports main
 echo "deb [arch=amd64] http://security.ubuntu.com/ubuntu $DIST-security main restricted universe multiverse" >> /etc/apt/sources.list
 env DEBIAN_FRONTEND=noninteractive apt-get update -y -o Acquire::Languages=none -o Acquire::Translation=none || true
 env DEBIAN_FRONTEND=noninteractive apt-get install -y gnupg
+
+set +e
+env DEBIAN_FRONTEND=noninteractive apt-get install -y curl
+curl -L https://dist.apache.org/repos/dist/dev/arrow/KEYS | apt-key add -
+apt-get install -y software-properties-common
```
Comment (tchaikov) [inline scripts/build_utils.sh:496]: could you please add `curl` and `software-properties-common` to `EXTRAPACKAGES`? see https://github.com/ceph/ceph-build/blob/297f54a86ed061fafdfebcefdc919980c34c848f/scripts/build_utils.sh#L604

**Review by tchaikov — COMMENTED** [review]
again, see https://github.com/ceph/ceph-build/pull/1827#pullrequestreview-685049111

File: scripts/build_utils.sh (hunk 1 above)
Comment (galsalomon66) [reply scripts/build_utils.sh]: :+1:

File: scripts/build_utils.sh [hunk 2]
```diff
@@ -601,7 +606,12 @@ setup_pbuilder() {
     fi
 
     if [ $FLAVOR = "crimson" ]; then
-        extrapackages='EXTRAPACKAGES="libc-ares-dev libcrypto++-dev libgnutls28-dev libhwloc-dev libnuma-dev libpciaccess-dev libprotobuf-dev libsctp-dev libyaml-cpp-dev protobuf-compiler ragel systemtap-sdt-dev"'
+        extrapackages='EXTRAPACKAGES="libc-ares-dev libcrypto++-dev libgnutls28-dev libhwloc-dev libnuma-dev libpciaccess-dev libprotobuf-dev libsctp-dev libyaml-cpp-dev protobuf-compiler ragel systemtap-sdt-dev software-properties-common curl"'
+        echo "$extrapackages" >> ~/.pbuilderrc
+    fi
+
+    if [ $FLAVOR = "default" ]; then
+        extrapackages='EXTRAPACKAGES="software-properties-common curl"'
```
Comment (tchaikov) [inline scripts/build_utils.sh:614]: i'd suggest avoid repeating `"software-properties-common curl"` at multiple places. that's error prune and difficult to maintain.

and i'd suggest avoid hardwire to "default" here. as **all** flavors should have these packages installed.

File: scripts/build_utils.sh [hunk 3]
```diff
@@ -600,10 +603,13 @@ setup_pbuilder() {
...
 
+    #in case it needs to update/add repo, the following packages should installed.
+    packages_for_repo_update="software-properties-common curl"
+
     if [ $FLAVOR = "crimson" ]; then
-        extrapackages='EXTRAPACKAGES="libc-ares-dev libcrypto++-dev libgnutls28-dev libhwloc-dev libnuma-dev libpciaccess-dev libprotobuf-dev libsctp-dev libyaml-cpp-dev protobuf-compiler ragel systemtap-sdt-dev"'
-        echo "$extrapackages" >> ~/.pbuilderrc
+        extrapackages="EXTRAPACKAGES=\"libc-ares-dev libcrypto++-dev libgnutls28-dev libhwloc-dev libnuma-dev libpciaccess-dev libprotobuf-dev libsctp-dev libyaml-cpp-dev protobuf-compiler ragel systemtap-sdt-dev ${packages_for_repo_update}\""
     fi
+    echo "$extrapackages" >> ~/.pbuilderrc
```
Comment (tchaikov) [inline scripts/build_utils.sh:612]:
```bash
extrapackages="software-properties-common curl"
if [ $FLAVOR = "crimson" ]; then
  extrapackages += " software-properties-common curl"
fi
echo "EXTRAPACKAGES=${extrapackages}" >> ~/.pbuilderrc
```

as discussed offline. i think we should enable "software-properties-common curl" for the flavors which are not "crimson". while your change does not take this into consideration.

**Review by tchaikov — APPROVED** [review]

**Comment by djgalloway** [comment]
This broke the CI.

```
E: Malformed entry 6 in list file /etc/apt/sources.list (Component)
E: The list of sources could not be read.
I: unmounting dev/ptmx filesystem
I: unmounting dev/pts filesystem
I: unmounting dev/shm filesystem
I: unmounting proc filesystem
I: unmounting sys filesystem
I: cleaning the build env 
I: removing directory /var/cache/pbuilder/build/938171 and its subdirectories
+ rm -fr /tmp/install-deps.937760
Build step 'Execute shell' marked build as failure
```

https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=focal,DIST=focal,MACHINE_SIZE=gigantic/55704//consoleFull

https://github.com/ceph/ceph-build/pull/1861

**Pull request merged**